### ISSUE TITLE
Cannot market buy indexes

### ISSUE THREAD

**Issue by jatinw21** [stabacco/stake-python#153]
```
async def weekly_market_buy(symbol="SPY", amount=100):
    async with stake.StakeClient() as stake_session:
        return await stake_session.trades.buy(
            stake.MarketBuyRequest(symbol=symbol, amount_cash=amount)
        )
```

I'm using this code for buying the symbol `SPY` on stake which is `S&P 500 ETF Trust SPDR` but it gives the following error

```
Traceback (most recent call last):
  File "/Users/.../Documents/Personal/personal-finance/stake/main.py", line 7, in <module>
    asyncio.run(weekly_market_buy('SPY', 10.0))
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/base_events.py", line 641, in run_until_complete
    return future.result()
  File "/Users/.../Documents/Personal/personal-finance/stake/DCA.py", line 10, in weekly_market_buy
    return await stake_session.trades.buy(
  File "/Users/.../Documents/Personal/personal-finance/env/lib/python3.10/site-packages/stake/trade.py", line 246, in buy
    return await self._trade(self._client.exchange.quick_buy, request)
  File "/Users/.../Documents/Personal/personal-finance/env/lib/python3.10/site-packages/stake/trade.py", line 188, in _trade
    product = await self._client.products.get(request_dict.pop("symbol"))
  File "/Users/.../Documents/Personal/personal-finance/env/lib/python3.10/site-packages/stake/product.py", line 74, in get
    return Product(**data["products"][0]) if data["products"] else None
  File "pydantic/main.py", line 342, in pydantic.main.BaseModel.__init__
pydantic.error_wrappers.ValidationError: 1 validation error for Product
id
  uuid version 4 expected (type=value_error.uuid.version; required_version=4)
```

The exact same code works perfectly fine when I run it for a symbol like 'AMZN'. So, the problem seems specific to this symbol or index symbols in general.

**Comment by jatinw21** [comment]
So apparently the client cannot find the product id from this symbol for some reason. I was able to resolve it by making a few changes to the trade class.

Firstly, I added an optional `item_id` parameter in `MarketBuyRequest` class

```python3
class MarketBuyRequest(BaseModel):
    """This is the request that needs to be passed to the trade client in order
    to buy some equity."""

    # AAPL, MSFT, TSLA etc...
    symbol: str
    amount_cash: float
    comments: Optional[str]

    item_id: Optional[str]
    item_type: str = "instrument"
    order_type: TradeType = TradeType.MARKET

    class Config:
        alias_generator = camelcase
        allow_population_by_field_name = True
```

Then I made a change in the `TradeClient` to use the item_id if provided and otherwise just use the symbol to get the item_id as normal.

```
class TradesClient(BaseClient):
    """This client is used to buy/sell equities."""

    async def _trade(
        self,
        url: str,
        request: Union[
            MarketBuyRequest,
            LimitBuyRequest,
            StopBuyRequest,
            LimitSellRequest,
            StopSellRequest,
            MarketSellRequest,
        ],
        check_success: bool = True,
    ) -> TradeResponse:
        """A generic function used to submit a trade, either buy or sell.

        Args:
            url: the url for buy/sell
            request:
            check_success: if true, an extra check will be performed to see if
            the transaction really occurred. This should not be needed, since it should
            be possible to inspect the TradeResponse for errors, but it does not look
            like they are recorded anywhere.

        Returns:
            the TradeResponse

        Raises:
            RuntimeError
        """
        request_dict = request.dict(by_alias=True)

        if "itemId" not in request_dict:
            product = await self._client.products.get(request_dict.pop("symbol"))
            assert product
            request_dict["itemId"] = str(product.id)
        else:
            request_dict.pop("symbol")

        request_dict["orderType"] = request_dict["orderType"].value
        request_dict["userId"] = str(self._client.user.id)
        
        data = await self._client.post(url, request_dict)
        trade = TradeResponse(**data[0])

        if check_success:
            await self._verify_successful_trade(trade)

        return trade
```

**Comment by stabacco** [comment]
Thanks for reporting this, I'll take a look shortly.

**Comment by stabacco** [comment]
Looks like the productId is of type `uuid1` instead of `uuid4` in the case of SPY. the stake api throws these kind of curve balls from time to time. #154 should fix it.

**Comment by stabacco** [comment]
Should be fixed in release 0.6.3. 
Please try that one out.

**Comment by jatinw21** [comment]
Thank you! Will check this out soon
